crates/gossip/proto/gossip.proto

```diff
@@ -4,46 +4,133 @@ package kitsune2.gossip;
 
 // A Kitsune2 gossip protocol message.
 message K2GossipMessage {
-  oneof gossip_message {
+  enum GossipMessageType {
+    UNSPECIFIED = 0;
+
     // A gossip initiation protocol message.
-    K2GossipInitiateMessage initiate = 1;
+    INITIATE = 1;
 
     // A gossip acceptance protocol message.
-    K2GossipAcceptMessage accept = 2;
+    ACCEPT = 2;
+
+    // A gossip no diff protocol message.
+    NO_DIFF = 3;
+
+    // A gossip agents protocol message.
+    AGENTS = 4;
   }
+
+  // The type of this message.
+  GossipMessageType msg_type = 1;
+
+  // The payload or content of this message.
+  bytes data = 2;
+}
+
+// A message representation of a Kitsune2 DHT arc set.
+message ArcSetMessage {
+  // The covered DHT sectors.
+  repeated uint32 arc_sectors = 1;
+}
+
+// A message representation of a Kitsune2 agent info.
+message AgentInfoMessage {
+  // The encoded agent info, which is signed by the provided signature
+  bytes encoded = 1;
+
+  // The signature of the encoded agent info.
+  bytes signature = 2;
 }
 
 // A Kitsune2 gossip initiation protocol message.
+//
+// Acceptable responses:
+// - `K2GossipAcceptMessage`
 message K2GossipInitiateMessage {
+  bytes session_id = 1;
+
   // The agent ids of the agents from the initiator who are in the in the peer store for the space where gossip is running.
-  repeated bytes participating_agents = 1;
+  repeated bytes participating_agents = 10;
```

Comment:
Why this jump to 10 and 20 indices?

Reply:
To leave room for new fields and to allow grouping fields together here. This is how I'm used to working with protobuf, doesn't have to stay

Reply:
Fine by me.